packages/cli/src/zosjobs/delete/old-jobs/OldJobs.handler.ts

```diff
@@ -0,0 +1,82 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { IHandlerParameters } from "@zowe/imperative";
+import { asyncPool } from "@zowe/core-for-zowe-sdk";
+import { DeleteJobs, GetJobs, IJob, JobsConstants, JOB_STATUS } from "@zowe/zos-jobs-for-zowe-sdk";
+import { ZosmfBaseHandler } from "@zowe/zosmf-for-zowe-sdk";
+
+/**
+ * "zos-jobs delete old-jobs" command handler. Delete (purge) multiple jobs in OUTPUT status.
+ * @export
+ * @class OldJobsHandler
+ * @implements {ICommandHandler}
+ */
+export default class OldJobsHandler extends ZosmfBaseHandler {
+    /**
+     * Command line arguments passed
+     * @private
+     * @type {*}
+     * @memberof OldJobsHandler
+     */
+    private arguments: any;
+
+    /**
+     * Command handler process - invoked by the command processor to handle the "zos-jobs delete job"
+     * @param {IHandlerParameters} params - Command handler parameters
+     * @returns {Promise<void>} - Fulfilled when the command completes successfully OR rejected with imperative error
+     * @memberof OldJobsHandler
+     */
+    public async processCmd(params: IHandlerParameters): Promise<void> {
+        this.arguments = params.arguments;
+
+        // Retrieve the list of user's jobs
+        const prefix: string = this.arguments.prefix || JobsConstants.DEFAULT_PREFIX;
+        const jobs: IJob[] = await GetJobs.getJobsByPrefix(this.mSession, prefix);
+
+        // Handle no jobs
+        if (jobs.length === 0) {
+            let notFoundMessage: string = "No jobs found";
+            if (prefix != null && prefix.length > 0) {
+                notFoundMessage += ` with prefix ${prefix}`;
+            }
+            this.console.log(notFoundMessage);
+            return;
+        }
+
+        // Loop through the jobs and delete those in OUTPUT status
+        const deletedJobs: IJob[] = [];
+        for (const job of jobs) {
+            if (job.status === JOB_STATUS.OUTPUT) {
+                deletedJobs.push(job);
+            }
+        }
+        const deleteJobPromise = (job: IJob) => DeleteJobs.deleteJobForJob(this.mSession, job);
+        if (this.arguments.maxConcurrentRequests === 0) {
+            await Promise.all(deletedJobs.map(deleteJobPromise));
+        } else {
+            await asyncPool(this.arguments.maxConcurrentRequests, deletedJobs, deleteJobPromise);
+        }
```

Comment:
I love the use of `Promise.all(...)` here 😋 
Question, should we get the API response and provide that as part of the `data.setObj` ?
Not sure if someone might be interesting in scripting based off of the JSON response coming back from this command.

Reply:
I wanted to implement it this way, as returning the API response from the `DeleteJobs` API would seem better than from the `GetJobs` API. However, the default version of the `modifyVersion` parameter here is `1.0` meaning that the z/OSMF request to delete the job is asynchronous so the API response is undefined.

This is how it was implemented in the original plug-in. I don't know if we'd want to use `modifyVersion = 2.0` instead to obtain an API response, or if that would have drawbacks like requiring a minimum z/OSMF version?

On the other hand, if the z/OSMF request to delete the job is asynchronous, does that mean we don't need to provide a `--max-concurrent-requests` option ourselves? I think it may still be beneficial as I did see some performance improvements with it.